core/src/main/java/io/dekorate/kubernetes/decorator/ResourceProvidingDecorator.java

```diff
@@ -51,22 +51,32 @@ public boolean match(HasMetadata h, String apiVersion, String kind, String name)
     return true;
   }
 
-  public Optional<ObjectMeta> getDeploymentMetadata(KubernetesListBuilder list) {
-    return list.getItems().stream().filter(h -> DEPLOYMENT_KINDS.contains(h.getKind())).map(HasMetadata::getMetadata)
-        .findFirst();
+  public List<ObjectMeta> getDeploymentMetadataList(KubernetesListBuilder list, String name) {
+    String trimedName = Strings.isNotNullOrEmpty(name) ? name.replaceAll("[:].*$", "") : name;
```

Comment:
I don't see the purpose of this replacement `name.replaceAll("[:].*$", "")`. I think it's new in this PR. I think adding a comment would be helpful to understand this rule.